### DISCUSSION TITLE
Error - Redirecting loop using SSO SAML authentication for a client app.

### DISCUSSION THREAD

**Discussion by keiken-shin** [keycloak/keycloak#28623] [Q&A]
Greetings,

I have a Cesium ION (self-hosted) solution deployed on kubernetes, and I was trying to add SSO SAML authentication to it. For this I came across Keycloak and set it up on Kubernetes.
I started following this [example](https://github.com/PabloLec/sb_keycloak_saml/tree/main?tab=readme-ov-file) provided in one of the discussions here. 
I imported the `IdP_Realm`, `SP_Realm` and `internal` json file to my keycloak and modified the URL according to the ones mentioned below.

**Below is my setup configuration:**

- Keycloak - `https://keycloak.threye.com`
- Cesium-ION (Client) - `https://ion.threye.com`
- Cesium SSO Configuration: 
```value.yaml
authenticationMode: saml
  saml:
    certificateSecret:
      name: "saml-secret"
      key: "saml.pem"
    # The SAML URL for your identity provider
    loginRequestUrl: "https://keycloak.threye.com/realms/SP_realm/protocol/openid-connect/auth?client_id=frontend&response_type=code&redirect_uri=https://ion.threye.com?kc_idp_hint=saml"
    # The entity ID that was configured in your identity provider when setting up the SAML application
    entityId: "https://keycloak.threye.com/realms/SP_realm"
    # The name ID format to use. Valid values are email, persistent or unspecified.
    nameIdFormat: "persistent"
    # Access to the Cesium ion Administrator user interface is granted to any
    # identity that matches the below name and value criteria
    # The attribute name to look up. Examples: "roles", "Groups"
    adminAttributeName: ""
    # The value that is expected to be found for adminAttributeName.
    # Examples: "admin", "admin-group", "superusers"
    adminAttributeValue: ""
```

Now when I start the Cesium app it redirects me to Keycloak Login page and then when I hit "Sign In" it keeps bouncing back and forth from ion to keycloak. I am unable to figure out why it's happening.  Please, if anyone could help me out here.

I have added a screen capture for the redirection loop error below:
https://github.com/keycloak/keycloak/assets/37549611/2aa8fe7f-0363-45a9-abd9-9ec1ce231d7d

Thanks!

**Comment by jsorah** [comment]
Your `loginRequestUrl` in your configuration looks suspect as that would be for an OIDC client. You mention you're trying to integrate a SAML client, so the SAML endpoint should be used. You can refer to the admin guide on this : https://www.keycloak.org/docs/latest/server_admin/#keycloak-server-saml-uri-endpoints

**Reply by keiken-shin** [reply]
Thank you for the response. I was unable to figure it out at that moment so switched to different SSO provider.  Also, the `loginRequestUrl` does contain a query flag called `kc_idp_hint=saml`, but I also think that's not the correct way to specify it's a SAML request and it's also not a correct SAML endpoint. I remember using SAML endpoint but that didn't worked out. Will give it a try again.
